cmd/object-api-utils_test.go

```diff
@@ -59,11 +61,14 @@ func TestIsValidBucketName(t *testing.T) {
 		{"ends-with-a-dot.", false},
 		{"ends-with-a-dash-", false},
 		{"-starts-with-a-dash", false},
-		{"THIS-BEINGS-WITH-UPPERCASe", false},
+		{"THIS-BEGINS-WITH-UPPERCASe", false},
 		{"tHIS-ENDS-WITH-UPPERCASE", false},
-		{"ThisBeginsAndEndsWithUpperCase", false},
+		{"ThisBeginsAndEndsWithUpperCasE", false},
 		{"una ñina", false},
-		{"lalalallalallalalalallalallalala-theString-size-is-greater-than-64", false},
+		{"dash-.may-not-appear-next-to-dot", false},
+		{"dash.-may-not-appear-next-to-dot", false},
+		{"dash-.-may-not-appear-next-to-dot", false},
```

Comment:
> When users try to mirror objects from a Minio server to AWS S3, the operation would fail with invalid bucket or object name, as the case may be. I think this is an important reason to fix it now.

Yeah agreed we don't need to defer, since we defer for all other invalid bucket names.